az-core/src/main/java/azkaban/Constants.java

```diff
@@ -883,11 +883,13 @@ public static class FlowParameters {
 
     // Constant to define the strategy to restart the execution, default to "restart_from_root",
     public static final String FLOW_PARAM_RESTART_STRATEGY = "flow.retry.strategy";
+  }
 
-    // default restart strategy, restart the execution in a whole, fresh new
-    public static final String FLOW_RESTART_STRATEGY_DEFAULT = "default";
+  public enum FlowRetryStrategy {
 
-    public static final String FLOW_RESTART_STRATEGY_DISABLE_SUCCEEDED_NODES =
-        "disable_succeeded_nodes";
+    // default restart strategy, restart the execution in a whole, fresh new
+    DEFAULT,
+    // restart the execution but disabling the corresponding succeeded nodes in previous execution
+    DISABLE_SUCCEEDED_NODES;
```

Comment:
What value do we want users to set for `flow.retry.strategy` property? 
1-DISABLE_SUCCEEDED_NODES
2-disable_succeeded_nodes
3-disableSucceededNodes - aligns with the naming pattern used by other value properties like finishCurrent, cancelImmediately, finishPossible

Also, for the DEFAULT retry strategy can we think of a more informative value? Thinking that value `default` may not explain much to users in terms of what will happen.